### PR TITLE
Add SPDX license headers to make project reuse-compliant

### PR DESCRIPTION
**Pull request by aesteve-rh**
Add SPDX headers to all files in the project and make
it reuse-compliant. The project will remain under the same
license (i.e., GPL-2.0-or-later) as before. Remove years to
ease maintenance and make the project consistent with
other projects in the organization (e.g., `Vdsm`).

Add `reuse lint` to the pipelines to ensure that the project
remains reuse-compliant.

Signed-off-by: Albert Esteve <aesteve@redhat.com>

### CODE REVIEW

File: go/imageio/extent.go [hunk 1]
```diff
@@ -37,7 +32,7 @@ type ExtentsResult interface {
 // ExtentsWraper wrapps []*Extent to provied the ExtentsResult interface.
 type ExtentsWrapper struct {
 	extents []*Extent
-	next int
+	next    int
```
Comment (nirs) [inline go/imageio/extent.go:35]: Unrelated change.

File: test/pki/system/ca.pem [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# SPDX-FileCopyrightText: Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+
```
Comment (nirs) [inline test/pki/system/ca.pem:3]: These files are generated by test/genpki. Generating new files will drop the headers
and fail the build. This is good enough for now, but we really need to add the license
headers automatically to the generated files, or use *.license files.

**Review by nirs — CHANGES_REQUESTED** [review]
Looks great, but one unrelated change need to be removed.

File: test/pki/system/ca.pem (hunk 2 above)
Comment (aesteve-rh) [reply test/pki/system/ca.pem:3]: I'll use *.license files then. It is better, I think.

File: go/imageio/backend.go [hunk 3]
```diff
@@ -1,10 +1,5 @@
-// ovirt-imageio
-// Copyright (C) 2021 Red Hat, Inc.
-//
-// This program is free software; you can redistribute it and/or modify
-// it under the terms of the GNU Lesser General Public License as
```
Comment (nirs) [inline go/imageio/backend.go:5]: The go/imageio directory is a library using the `GNU Lesser General Public License` (LPGL).

File: go/imageio/backend.go [hunk 4]
```diff
@@ -1,10 +1,5 @@
...
-// published by the Free Software Foundation; either version 2 of the
-// License, or (at your option) any later version.
+// SPDX-FileCopyrightText: Red Hat, Inc.
+// SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (nirs) [inline go/imageio/backend.go:2]: You replaced it with GPL!

I'm not sure if LGPL actually works with the way Go projects are built
but this commit should not change the license.

We need to add LGPL license to LICENCES/ and use the right spdx header
for this directory.

File: go/ovirt-img/backend.go [hunk 5]
```diff
@@ -1,10 +1,5 @@
-// ovirt-imageio
-// Copyright (C) 2021 Red Hat, Inc.
-//
-// This program is free software; you can redistribute it and/or modify
-// it under the terms of the GNU General Public License as published by
-// the Free Software Foundation; either version 2 of the License, or
-// (at your option) any later version.
+// SPDX-FileCopyrightText: Red Hat, Inc.
+// SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (nirs) [inline go/ovirt-img/backend.go:2]: The ovirt-img directory uses GPL, so changes in this directory are fine.

File: go/imageio/README.md [hunk 6]
```diff
@@ -58,4 +58,5 @@ Usage example:
 ## LICENSE
 
 The is free software licensed under the GNU Lesser General Public
-License version 2 or above (LGPLv2+). See the file LICENSE for details.
+License version 2 or above (GPL-2.0-or-later). See the file
+`LICENSES/GPL-2.0-or-later.txt` file in the project root for details.
```
Comment (nirs) [inline go/imageio/README.md:62]: This directory uses LGPL.

File: MANIFEST.in [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # SPDX-FileCopyrightText: Red Hat, Inc.
 # SPDX-License-Identifier: GPL-2.0-or-later
 
-include README.md COPYING MANIFEST
+include README.md LICENSES/* MANIFEST
```
Comment (nirs) [inline MANIFEST.in:4]: All python files use only GPL, so we really need to include only the GPL license.

File: ovirt-imageio.spec.in [hunk 8]
```diff
@@ -80,7 +80,7 @@ Requires:  python3
 Common resources used by oVirt imageio server and client
 
 %files common
-%license COPYING
+%license LICENSES/*
```
Comment (nirs) [inline ovirt-imageio.spec.in:83]: Same, we need to install only the GPL license.

File: ovirt-imageio.spec.in [hunk 9]
```diff
@@ -191,7 +191,7 @@ Requires:  python38
 Common resources used by oVirt imageio server and client
 
 %files -n python38-%{name}-common
-%license COPYING
+%license LICENSES/*
```
Comment (nirs) [inline ovirt-imageio.spec.in:194]: Same

**Review by nirs — CHANGES_REQUESTED** [review]
Looks fine except the unintended change from LGPL to GPL.

**Review by nirs — APPROVED** [review]
